stdlib/tempfile/0/tempfile.rbs

```diff
@@ -0,0 +1,267 @@
+  # use finalizer and delegation, which makes it more efficient and reliable.
+  #
+  # If no block is given, this is similar to Tempfile.new except creating File
+  # instead of Tempfile. In that case, the created file is not removed
+  # automatically. You should use File.unlink to remove it.
+  #
+  # If a block is given, then a File object will be constructed, and the block is
+  # invoked with the object as the argument. The File object will be automatically
+  # closed and the temporary file is removed after the block terminates, releasing
+  # all resources that the block created. The call returns the value of the block.
+  #
+  # In any case, all arguments (`basename`, `tmpdir`, `mode`, and `**options`)
+  # will be treated the same as for Tempfile.new.
+  #
+  #     Tempfile.create('foo', '/home/temp') do |f|
+  #        # ... do something with f ...
+  #     end
+  #
+  def self.create: (?String basename, ?String? tmpdir, ?mode: Integer, **untyped) -> File
+
+  # Creates a new Tempfile.
+  #
+  # This method is not recommended and exists mostly for backward compatibility.
+  # Please use Tempfile.create instead, which avoids the cost of delegation, does
+  # not rely on a finalizer, and also unlinks the file when given a block.
+  #
+  # Tempfile.open is still appropriate if you need the Tempfile to be unlinked by
+  # a finalizer and you cannot explicitly know where in the program the Tempfile
+  # can be unlinked safely.
+  #
+  # If no block is given, this is a synonym for Tempfile.new.
+  #
+  # If a block is given, then a Tempfile object will be constructed, and the block
+  # is run with the Tempfile object as argument. The Tempfile object will be
+  # automatically closed after the block terminates. However, the file will
+  # **not** be unlinked and needs to be manually unlinked with Tempfile#close! or
+  # Tempfile#unlink. The finalizer will try to unlink but should not be relied
+  # upon as it can keep the file on the disk much longer than intended. For
+  # instance, on CRuby, finalizers can be delayed due to conservative stack
+  # scanning and references left in unused memory.
+  #
+  # The call returns the value of the block.
+  #
+  # In any case, all arguments (`*args`) will be passed to Tempfile.new.
+  #
+  #     Tempfile.open('foo', '/home/temp') do |f|
+  #        # ... do something with f ...
+  #     end
+  #
+  #     # Equivalent:
+  #     f = Tempfile.open('foo', '/home/temp')
+  #     begin
+  #        # ... do something with f ...
+  #     ensure
+  #        f.close
+  #     end
+  #
+  def self.open: (*untyped args, **untyped) -> Tempfile
+
+  public
+
+  # Closes the file. If `unlink_now` is true, then the file will be unlinked
+  # (deleted) after closing. Of course, you can choose to later call #unlink if
+  # you do not unlink it now.
+  #
+  # If you don't explicitly unlink the temporary file, the removal will be delayed
+  # until the object is finalized.
+  #
+  def close: (?bool unlink_now) -> bool?
```

Comment:
> __[Goodcheck]__ Prefer `boolish` over `bool` for method arguments and block return values ([view](https://sider.review/gh/repos/174796313/pulls/767#issue-5983947))
>
> <details>
>
> See the doc below:
> https://github.com/ruby/rbs/blob/78d04a2db0f1c4925d2b13c2939868edf9465d6c/docs/syntax.md#bool-or-boolish
>
> | Rule |
> | ---- |
> | `rbs.prefer_boolish` |
>
> You can [close this issue](https://sider.review/gh/repos/174796313/pulls/767#issue-5983947) if no need to fix it. [Learn more](https://help.sider.review/getting-started/working-with-issues).
>
> </details>

Could you make the type of the parameter `boolish`?

Reply:
I tried but then got into this trouble
```
  def test_close
    assert_send_type "(?boolish unlink_now) -> bool?",
                     Tempfile.new('README.md'), :close

    assert_send_type "(?boolish unlink_now) -> bool?",
                     Tempfile.new('README.md'), :close, true

    assert_send_type "(?boolish unlink_now) -> bool?",
                     Tempfile.new('README.md'), :close, false
  end

```

```
Started
....E
Error: test_close(TempfileTest): RuntimeError: Unknown name for expand_alias: boolish
/rbs/lib/rbs/definition_builder.rb:784:in `expand_alias'
/rbs/lib/rbs/test/type_check.rb:319:in `value'
/rbs/lib/rbs/test/type_check.rb:79:in `block in args'
/rbs/lib/rbs/test/type_check.rb:178:in `block in zip_args'
<internal:kernel>:148:in `yield_self'
/rbs/lib/rbs/test/type_check.rb:176:in `zip_args'
/rbs/lib/rbs/test/type_check.rb:78:in `args'
/rbs/lib/rbs/test/type_check.rb:50:in `method_call'
/rbs/test/stdlib/test_helper.rb:265:in `assert_send_type'
/rbs/test/stdlib/tempfile/Tempfile_test.rb:54:in `test_close'
     51:     assert_send_type "(?boolish unlink_now) -> bool?",
     52:                      Tempfile.new('README.md'), :close
     53:
  => 54:     assert_send_type "(?boolish unlink_now) -> bool?",
     55:                      Tempfile.new('README.md'), :close, true
     56:
     57:     assert_send_type "(?boolish unlink_now) -> bool?",
=======================================================================================
........
Finished in 0.008122501 seconds.
---------------------------------------------------------------------------------------
13 tests, 34 assertions, 0 failures, 1 errors, 0 pendings, 0 omissions, 0 notifications
92.3077% passed
---------------------------------------------------------------------------------------
```
I'm not quite sure about the error here, please help me with this :pray:

Reply:
Yeah, you can write `::boolish`. But, I recommend writing the specific types of the arguments and return value.

```
  def test_close
    assert_send_type "() -> void",
                     Tempfile.new('README.md'), :close

    assert_send_type "(true) -> void",
                     Tempfile.new('README.md'), :close, true

    assert_send_type "(String) -> void",
                     Tempfile.new('README.md'), :close, "true"

    assert_send_type "(false) -> void",
                     Tempfile.new('README.md'), :close, false
  end
```

I think `(?boolish) -> void` would be the correct type here.

Reply:
I've updated the code. Thank you so much for your support :tada: